tuning/data/data_processors.py

```diff
@@ -46,9 +46,20 @@ def __init__(
         # Initialize other objects
         self.registered_handlers = {}
 
-    def register_data_handler(self, name: str, func: callable):
+    def register_data_handler(self, name: str, func: Callable):
+        if not isinstance(name, str) or not callable(func):
+            raise ValueError("Handlers should be of type Dict, str to callable")
+        logging.info("Registering handler %s passed by the user", name)
         self.registered_handlers[name] = func
```

Comment:
Do you want to return a warn message if a data handler of the same name exists before?

Reply:
added.